zerver/tests/test_cache.py

```diff
@@ -85,14 +85,15 @@ def get_user_function_with_bad_cache_keys(user_id: int) -> UserProfile:
 
         hamlet = self.example_user('hamlet')
         with patch('zerver.lib.cache.cache_set') as mock_set, \
-                patch('zerver.lib.cache.logger.warning') as mock_warn:
+                self.assertLogs(level="WARNING"):
             with queries_captured() as queries:
                 result = get_user_function_with_bad_cache_keys(hamlet.id)
 
+            # Logs not caught here as they involved a key value of which changed in every run,
+            # Previouly also the log out was not caught only logging did happen was checked.
             self.assertEqual(result, hamlet)
             self.assert_length(queries, 1)
             mock_set.assert_not_called()
-            mock_warn.assert_called_once()
```

Comment:
Can we replace `assert_called_once` by asserting if the object returned by `assertLogs` has `output` of length greater than 1? I think that would be a better choice.